xla/service/gpu/gemm_algorithm_picker.cc

```diff
@@ -242,21 +246,15 @@ class GemmAutotuner {
       return std::move(profile_result);
     };
 
-    TF_ASSIGN_OR_RETURN(best_algorithm,
-                        GetBestAlgorithm<se::blas::AlgorithmType>(
-                            gemm, algorithms, gemm_config.beta, tuned_func));
-    if (best_algorithm.has_gemm()) {
-      int alg_idx = best_algorithm.gemm().algorithm();
-      best_algorithm.mutable_gemm()->set_algorithm(algorithms[alg_idx]);
-    }
-    return best_algorithm;
+    return GetBestAlgorithm<se::blas::AlgorithmType>(
+                      gemm, algorithms, gemm_config.beta, false, tuned_func);
```

Comment:
I would really prefer to remove this parameter (see other comment). If it truly has to stay, then please add `/*return_algo_index=*/` in front.